app/src/main/java/com/cornellappdev/transit/util/IntentUtils.kt

```
@@ -0,0 +1,25 @@
+object IntentUtils {
+    fun Context.openDeepLink(packageName: String) {
+        val launchIntent = packageManager.getLaunchIntentForPackage(packageName)
+
+        if (launchIntent != null) {
+            startActivity(launchIntent)
+        } else {
+            try {
+                val playStoreIntent = Intent(Intent.ACTION_VIEW, "market://details?id=$packageName".toUri())
+                playStoreIntent.setPackage("com.android.vending")
+                startActivity(playStoreIntent)
+            } catch (e2: ActivityNotFoundException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)